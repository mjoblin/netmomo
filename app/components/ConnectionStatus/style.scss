.connection-status {
    .status-text {
        margin-left: .8em;
    }
}