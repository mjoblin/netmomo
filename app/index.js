import React from 'react';
import ReactDOM from 'react-dom';
import { AppContainer } from 'react-hot-loader';
import 'react-hot-loader/patch';

import Root from './containers/Root';
import store from './store';


const render = Component => {
    ReactDOM.render(
        <AppContainer>
            <Component store={store} />
        </AppContainer>,
        document.getElementById('root')
    );
};

render(Root);

if (module.hot) {
    module.hot.accept('./containers/Root', () => { render(Root) });
}
