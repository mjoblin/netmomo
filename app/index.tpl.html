<!doctype html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <title>NetDumpling Viewer</title>
    <meta name="description" content="NetDumpling Viewer">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
</head>

<body class="view-login">
<div id="root"></div>
</body>

</html>