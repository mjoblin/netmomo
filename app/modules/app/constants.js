export const NAME = 'app';

export const HUB_CONNECTING = 'Connecting';
export const HUB_CONNECTED = 'Connected';
export const HUB_DISCONNECTED = 'Disconnected';
export const HUB_RECONNECTING = 'Reconnecting';
