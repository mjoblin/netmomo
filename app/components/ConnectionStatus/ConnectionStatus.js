import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Switch } from 'antd';

import appModule from 'AppRoot/modules/app';
import settingsModule from 'AppRoot/modules/settings';
import './style.scss';


export const ConnectionStatus = class extends React.Component {
    render() {
        const { actions, settings } = this.props;

        return (
            <span className="connection-status">
                <Switch
                    checked={this.props.hubConnected}
                    onChange={
                        val => val ?
                            actions.hubConnect(settings.hubHost, settings.hubPort) :
                            actions.hubDisconnect()
                    }
                />
                <span className="status-text">
                    { this.props.hubConnectionStatus }
                </span>
            </span>
        );
    }
};

ConnectionStatus.displayName = 'ConnectionStatus';

ConnectionStatus.propTypes = {
    hubConnectionStatus: PropTypes.string.isRequired,
    hubConnected: PropTypes.bool.isRequired,
    settings: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
    hubConnectionStatus: appModule.selectors.hubConnectionStatus(state),
    hubConnected: appModule.selectors.hubConnected(state),
    settings: settingsModule.selectors.getSettings(state),
});

const mapDispatchToProps = dispatch => {
    return {
        actions: bindActionCreators({
            hubConnect: appModule.actions.hubConnect,
            hubDisconnect: appModule.actions.hubDisconnect
        }, dispatch)
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ConnectionStatus);
