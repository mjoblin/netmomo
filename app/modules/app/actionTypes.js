import { NAME } from './constants';


export const HUB_CONNECT = `${NAME}/HUB_CONNECT`;
export const HUB_DISCONNECT = `${NAME}/HUB_DISCONNECT`;

export const HUB_CONNECTING = `${NAME}/HUB_CONNECTING`;
export const HUB_UNAVAILABLE = `${NAME}/HUB_UNAVAILABLE`;
export const HUB_CONNECTED = `${NAME}/HUB_CONNECTED`;
export const HUB_DISCONNECTED = `${NAME}/HUB_DISCONNECTED`;
export const HUB_ERROR = `${NAME}/HUB_ERROR`;
export const HUB_RECONNECT_ATTEMPT = `${NAME}/HUB_RECONNECT_ATTEMPT`;
export const HUB_CANCEL_RECONNECT = `${NAME}/HUB_CANCEL_RECONNECT`;

export const DUMPLING = `${NAME}/DUMPLING`;
