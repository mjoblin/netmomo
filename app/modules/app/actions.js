import * as t from './actionTypes';


export const newDumpling = dumpling => ({
    type: t.DUMPLING,
    dumpling,
});

export const shiftyConnect = (host, port) => ({ type: t.SHIFTY_CONNECT, host, port });
export const shiftyDisconnect = () => ({ type: t.SHIFTY_DISCONNECT });

export const shiftyConnecting = () => ({ type: t.SHIFTY_CONNECTING });
export const shiftyUnavailable = () => ({ type: t.SHIFTY_UNAVAILABLE });
export const shiftyConnected = () => ({ type: t.SHIFTY_CONNECTED });
export const shiftyDisconnected = () => ({ type: t.SHIFTY_DISCONNECTED });
export const shiftyError = error => ({ type: t.SHIFTY_ERROR, error });
export const shiftyCancelReconnect = () => ({ type: t.SHIFTY_CANCEL_RECONNECT });

export const shiftyReconnectAttempt = delay => ({
    type: t.SHIFTY_RECONNECT_ATTEMPT,
    delay,
});