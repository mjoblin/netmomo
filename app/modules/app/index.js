import * as actions from './actions';
import * as actionTypes from './actionTypes';
import * as components from './components';
import * as constants from './constants';
import reducer from './reducer';
import sagas from './sagas';


export default { actions, actionTypes, components, constants, reducer, sagas };