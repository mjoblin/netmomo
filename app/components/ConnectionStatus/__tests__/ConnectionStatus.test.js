import React from 'react';
import { shallow } from 'enzyme';
import { ConnectionStatus } from '../ConnectionStatus';
import { HUB_CONNECTED } from "AppRoot/modules/app/constants";


describe('ConnectionStatus', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <ConnectionStatus
                hubConnectionStatus={HUB_CONNECTED}
                hubConnected={true}
                settings={{ hubHost: 'localhost', hubPort: 11348 }}
                actions={{}}
            />
        );

        expect(component).toMatchSnapshot();
    });
});