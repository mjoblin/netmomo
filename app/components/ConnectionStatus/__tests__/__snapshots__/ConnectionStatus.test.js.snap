// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatus matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConnectionStatus
    actions={Object {}}
    hubConnected={true}
    hubConnectionStatus="Connected"
    settings={
      Object {
        "hubHost": "localhost",
        "hubPort": 11348,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Switch
          checked={true}
          onChange={[Function]}
          prefixCls="ant-switch"
        />,
        <span
          className="status-text"
        >
          Connected
        </span>,
      ],
      "className": "connection-status",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "checked": true,
          "onChange": [Function],
          "prefixCls": "ant-switch",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Connected",
          "className": "status-text",
        },
        "ref": null,
        "rendered": "Connected",
        "type": "span",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Switch
            checked={true}
            onChange={[Function]}
            prefixCls="ant-switch"
          />,
          <span
            className="status-text"
          >
            Connected
          </span>,
        ],
        "className": "connection-status",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "checked": true,
            "onChange": [Function],
            "prefixCls": "ant-switch",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Connected",
            "className": "status-text",
          },
          "ref": null,
          "rendered": "Connected",
          "type": "span",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
